deploy/buildtools/bit_builders/vitis.py

```diff
@@ -0,0 +1,206 @@
+from __future__ import with_statement, annotations
+
+import logging
+import os
+from fabric.api import prefix, local, run # type: ignore
+from fabric.contrib.project import rsync_project # type: ignore
+
+from util.streamlogger import StreamLogger, InfoStreamLogger
+from buildtools.bitbuilder import BitBuilder
+
+# imports needed for python type checking
+from typing import Optional, Dict, Any, TYPE_CHECKING
+if TYPE_CHECKING:
+    from buildtools.buildconfig import BuildConfig
+
+rootLogger = logging.getLogger()
+
+def get_deploy_dir() -> str:
+    """Determine where the firesim/deploy directory is and return its path.
+
+    Returns:
+        Path to firesim/deploy directory.
+    """
+    with StreamLogger('stdout'), StreamLogger('stderr'):
+        deploydir = local("pwd", capture=True)
+    return deploydir
+
+class VitisBitBuilder(BitBuilder):
+    """Bit builder class that builds a Vitis bitstream from the build config.
+
+    Attributes:
+    """
+
+    def __init__(self, build_config: BuildConfig, args: Dict[str, Any]) -> None:
+        super().__init__(build_config, args)
+
+    def replace_rtl(self):
+        rootLogger.info(f"Building Verilog for {self.build_config.get_chisel_triplet()}")
+
+        with prefix(f'cd {get_deploy_dir()}/../'), \
```

Comment:
If we could put these prefixes in a reusable place we'd save 100s of lines across the codebase.

Reply:
I'm not sure about 100s but I agree with you. Unfortunately, I couldn't find anything by quickly Googling. We can do this after the release.